python/ray/serve/_private/logging_utils.py

```diff
@@ -152,7 +154,11 @@ def access_log_msg(*, method: str, status: str, latency_ms: float):
 
 
 def log_to_stderr_filter(record: logging.LogRecord) -> bool:
-    """Filters log records based on a parameter in the `extra` dictionary."""
+    """Filters log records based on a parameter in the `extra` dictionary or
+    RAY_SERVE_DISABLE_STDOUT environment variable."""
+    if RAY_SERVE_DISABLE_STDOUT:
+        return False
```

Comment:
A slightly more optimal way to do this would be to avoid adding the stderr stream handler entirely when the env var is set, but I don't have a strong preference.

Reply:
oh let me try